File: services/vscode/test/unit/jwt.test.ts
import { describe, expect, it } from "vitest";
import { decodeJwtPayload } from "../../src/auth/jwt";
const b64 = (o: unknown) => Buffer.from(JSON.stringify(o)).toString("base64url");
describe("decodeJwtPayload", () => {
  it("decodes the payload segment", () => {
    const tok = `${b64({ alg: "HS256" })}.${b64({ sub: "u1", email: "a@b", role: "operator", is_superadmin: false, type: "access", exp: 1 })}.sig`;
    expect(decodeJwtPayload(tok)).toMatchObject({ email: "a@b", role: "operator", is_superadmin: false });
  });
  it("returns undefined for garbage / API keys", () => {
    expect(decodeJwtPayload("tdt_abc")).toBeUndefined();
    expect(decodeJwtPayload("a.b")).toBeUndefined();
  });
});

File: services/jetbrains/src/test/kotlin/com/terraducktel/jetbrains/auth/JwtTest.kt
package com.terraducktel.jetbrains.auth

import org.junit.Assert.*
import org.junit.Test
import java.util.Base64

/** Port of `services/vscode/test/unit/jwt.test.ts`. */
class JwtTest {
    private fun b64(json: String): String = Base64.getUrlEncoder().withoutPadding().encodeToString(json.toByteArray(Charsets.UTF_8))

    @Test fun `decodes the payload segment`() {
        val header = b64("""{"alg":"HS256"}""")
        val payload = b64("""{"sub":"u1","email":"a@b","role":"operator","is_superadmin":false,"type":"access","exp":1}""")
        val token = "$header.$payload.sig"

        val claims = Jwt.decodePayload(token)

        assertNotNull(claims)
        assertEquals("a@b", claims!!.email)
        assertEquals("operator", claims.role)
        assertEquals(false, claims.is_superadmin)
    }

    @Test fun `returns null for garbage and API keys`() {
        assertNull(Jwt.decodePayload("tdt_abc"))
        assertNull(Jwt.decodePayload("a.b"))
    }
}
